src/data/fin/tuple/sort.lean

```diff
@@ -91,3 +90,104 @@ begin
 end
 
 end tuple
+
+namespace tuple
+
+open list
+
+variables {n : ℕ} {α : Type*}
+
+/-- If we swap two strictly decreasing entries in a tuple, then the result is lexicographically
+smaller than the original tuple. -/
+lemma lex_desc [preorder α] {f : fin n → α} {i j : fin n} (h₁ : i < j) (h₂ : f j < f i) :
+  to_lex (f ∘ equiv.swap i j) < to_lex f :=
```

Comment:
Yes:
```lean
lemma lex_desc' {ι α} [preorder ι] [decidable_eq ι] [preorder α] {f : ι → α} {i j : ι}
  (h₁ : i < j) (h₂ : f j < f i) :
  to_lex (f ∘ equiv.swap i j) < to_lex f :=
```
with the same proof. Should I use this version? Should it then still be called `tuple.lex_desc` or go somewhere else and get a different name prefix?

Reply:
Yes I think it fits in the pi/lex file; I imagine logic/equiv/basic (where `equiv.swap` is defined) should already be imported, but in case not it seems totally appropriate to import such a basic file. (If you want to add the `is_asymm` + `r i j` version: you may also add a `is_irrefl` + ¬ r j i` version.)

Reply:
Not sure about `pi`, since the target type of `f` cannot be dependent if we swap things in the domain.

Reply:
Yes you can only state it for `f : ι → α` but the lex order on `ι → α` is just a specialization of `pi.lex` so you can only add it after `pi.lex` (and the (<) instance on `lex (Π i, β i)`) is defined, right?

Reply:
I guess so. So I'll move it to the end of `data,pi.lex` as `pi.lex_desc`. But I don't think I'll cook up more general versions (for non-preorders on the indexing type) for now. These PRs have a tendency of growing tentacles, a bit like the hydra game... :smile: